action.yaml

```diff
@@ -52,4 +52,4 @@ runs:
       - name: Install Dependencies
         shell: bash
         run: |
-          pip install -r requirements.txt -r requirements-dev.txt
+          pip install --no-deps -r requirements-dev.txt
```

Comment:
I like your PR description, maybe we can just leave it here as a comment to document why we are using the --no-deps flag.

Reply:
I mean specifically this part:

```
The pip install should use --no-deps to prevent it from trying to install package dependencies for two reasons:

The canonical set of dependencies to install for a given service should come from the lock files and the lock files only.
If pip does look at package dependencies when installing a given package (like FastAPI), there is a potential for false conflicts to arise. For example, FastAPI requires Pydantic with a specification that is (according to pip) invalid when requiring hashes. The Pydantic dependency is already pinned in our lock file and pip has no need to try to resolve FastAPI's dependencies.
```